> ```powershell
> 2025-07-11T08:58:41.9387126Z Caused by: java.lang.NullPointerException: Cannot invoke "com.mongodb.ConnectionString.getConnectionString()" because "this.connectionString" is null
> 2025-07-11T08:58:41.9388198Z 	at liquibase.ext.mongodb.database.MongoConnection.getVisibleUrl(MongoConnection.java:142)
> 2025-07-11T08:58:41.9389000Z 	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:75)
> 2025-07-11T08:58:41.9389765Z 	at liquibase.command.core.UpdateCommandStep.run(UpdateCommandStep.java:100)
> 2025-07-11T08:58:41.9390417Z 	at liquibase.command.CommandScope.lambda$execute$6(CommandScope.java:263)
> 2025-07-11T08:58:41.9391101Z 	... 22 more
> ```

@geoand thanks for tagging me, hold on for liquibase-mongodb 4.33.0.1 still today including this fix.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -162,8 +162,8 @@
         <flyway.version>11.10.2</flyway.version>
         <yasson.version>3.0.4</yasson.version>
         <!-- liquibase-mongodb is not released everytime with liquibase anymore, but the two versions need to be compatible -->
-        <liquibase.version>4.29.1</liquibase.version>
-        <liquibase-mongodb.version>4.29.1</liquibase-mongodb.version>
+        <liquibase.version>4.33.0</liquibase.version>
+        <liquibase-mongodb.version>4.33.0.1</liquibase-mongodb.version>
         <snakeyaml.version>2.4</snakeyaml.version>
         <osgi.version>6.0.0</osgi.version>
         <mongo-client.version>5.5.1</mongo-client.version>

--- a/extensions/liquibase/liquibase-common/src/main/java/io/quarkus/liquibase/common/runtime/graal/SubstituteScope.java
+++ b/extensions/liquibase/liquibase-common/src/main/java/io/quarkus/liquibase/common/runtime/graal/SubstituteScope.java
@@ -1,0 +1,39 @@
+package io.quarkus.liquibase.common.runtime.graal;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.RecomputeFieldValue;
+import com.oracle.svm.core.annotate.TargetClass;
+
+import liquibase.Scope;
+import liquibase.ScopeManager;
+import liquibase.SingletonScopeManager;
+
+@TargetClass(value = Scope.class)
+public final class SubstituteScope {
+
+    /**
+     * All the following code is here to reset <a href=
+     * "https://github.com/liquibase/liquibase/blob/51de1de4437e5b5fbcbd25cff006d1c6d5313bab/liquibase-standard/src/main/java/liquibase/Scope.java#L95-L102">this</a>
+     */
+
+    @Alias
+    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.NewInstance, declClass = CustomInheritableThreadLocal.class)
+    private static InheritableThreadLocal<ScopeManager> scopeManager;
+
+    public static final class CustomInheritableThreadLocal extends InheritableThreadLocal<ScopeManager> {
+        @Override
+        protected ScopeManager childValue(ScopeManager parentValue) {
+            CustomScopeManager sm = new CustomScopeManager();
+            sm.setCurrentScope(parentValue.getCurrentScope());
+            return sm;
+        }
+    }
+
+    public static class CustomScopeManager extends SingletonScopeManager {
+        @Override
+        public synchronized void setCurrentScope(Scope scope) {
+            super.setCurrentScope(scope);
+        }
+    }
+
+}

--- a/extensions/liquibase/liquibase-mongodb/deployment/src/main/java/io/quarkus/liquibase/mongodb/deployment/LiquibaseMongodbProcessor.java
+++ b/extensions/liquibase/liquibase-mongodb/deployment/src/main/java/io/quarkus/liquibase/mongodb/deployment/LiquibaseMongodbProcessor.java
@@ -111,6 +111,12 @@ class LiquibaseMongodbProcessor {
         reflective.produce(ReflectiveClassBuildItem.builder(
                 liquibase.change.AbstractSQLChange.class.getName(),
                 liquibase.ext.mongodb.change.AbstractMongoChange.class.getName(),
+                liquibase.ext.mongodb.change.CreateCollectionChange.class.getName(),
+                liquibase.ext.mongodb.change.CreateIndexChange.class.getName(),
+                liquibase.ext.mongodb.change.DropCollectionChange.class.getName(),
+                liquibase.ext.mongodb.change.DropIndexChange.class.getName(),
+                liquibase.ext.mongodb.change.InsertManyChange.class.getName(),
+                liquibase.ext.mongodb.change.InsertOneChange.class.getName(),
                 liquibase.database.jvm.JdbcConnection.class.getName())
                 .methods().build());
 

--- a/extensions/liquibase/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java
+++ b/extensions/liquibase/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java
@@ -230,6 +230,11 @@ class LiquibaseProcessor {
                     .constructors().methods().fields().build());
         });
 
+        reflective.produce(ReflectiveClassBuildItem.builder(
+                liquibase.precondition.PreconditionLogic.class.getName())
+                .reason(getClass().getName())
+                .fields().build());
+
         var dependencies = curateOutcome.getApplicationModel().getRuntimeDependencies();
         resource.produce(NativeImageResourceBuildItem.ofDependencyResources(
                 dependencies, LIQUIBASE_ARTIFACT, LIQUIBASE_RESOURCE_FILTER));

